core/src/test/java/net/sourceforge/jnlp/services/SingleInstanceLockTest.java

```diff
@@ -0,0 +1,38 @@
+package net.sourceforge.jnlp.services;
+
+import java.net.URL;
+
+import org.junit.Test;
+
+import net.adoptopenjdk.icedteaweb.io.FileUtils;
+import net.sourceforge.jnlp.JNLPFile;
+import net.sourceforge.jnlp.ParserSettings;
+import net.sourceforge.jnlp.util.logging.NoStdOutErrTest;
+
+public final class SingleInstanceLockTest
+	extends NoStdOutErrTest
+{
+    @Test
+    public void testCreateWithPort() 
+    	throws Exception 
+    {
+    	final URL url = this.getClass().getClassLoader().getResource("net/sourceforge/jnlp/basic.jnlp");
+        final JNLPFile jnlpFile = new JNLPFile(url.openStream(), url, new ParserSettings(false,false,false));
+        
+        assert jnlpFile!=null;
+        
+        final SingleInstanceLock sil = new SingleInstanceLock(jnlpFile);
+        
+        sil.lockFile.delete();
+        assert sil.lockFile.exists()==false;
+        
+        sil.createWithPort(123);
+        assert sil.lockFile.exists()==true;
+        assert "123".equals(FileUtils.loadFileAsUtf8String(sil.lockFile).trim());
+        
+
+        sil.createWithPort(456);

```

Comment:
Should this really be possible? I would assume that a lock must be removed before a new one can be created.

Reply:
Icedtea-web does not remove lock file neither before application start nor after application termination. At least i was not able to find this is source code.

Reply:
mmh, ok. I would assume that creating a lock will create a file and remove a lock will delete the file. Creating a without deleting a previous lock should throw an exception. But I will have a deeper look.

Reply:
The naming of this is bad.
The lock file is not used for locking but for broadcasting a port.

This port is then used as
- a lock because only one process can listen on a port
- a communication channel to inform the running application that a second instance should be started